app/components/topbar/WalletTopbarTitle.js

```diff
@@ -81,10 +88,30 @@ export default class WalletTopbarTitle extends Component<Props> {
         </div>
         <div className={styles.divAmount}>
           <div className={styles.walletAmount}>
-            { wallet && formattedWalletAmount(wallet.amount) + ' ADA' }
+            { wallet && shouldHideBalance ?
+              <span className={styles.hiddenWalletAmount}>
+                {formattedWalletAmount(wallet.amount).replace(/./g, '*')}
```

Comment:
do this leaks the order of magnitude?

Reply:
Yes. Would you prefer a constant-length, asterisk string? I think it'd be better too.

Reply:
Yeah, I think so too